transport/http/outbound.go

```diff
@@ -323,6 +339,13 @@ func (o *Outbound) withOpentracingSpan(ctx context.Context, req *http.Request, t
 }
 
 func (o *Outbound) withCoreHeaders(req *http.Request, treq *transport.Request, ttl time.Duration) *http.Request {
+	// Add default headers to all requests.
+	for k, vs := range o.headers {
+		for _, v := range vs {
+			req.Header.Add(k, v)
```

Comment:
Might be worth it - that would certainly would be alot more transparent than overwriting their headers.